Remotion/Data/DomainObjects.UnitTests/Persistence/Rdbms/SqlServer/Parameters/SqlDataParameterDefinitionFactoryDecoratorTests.cs

```diff
@@ -0,0 +1,44 @@
+// This file is part of the re-motion Core Framework (www.re-motion.org)
+// Copyright (c) rubicon IT GmbH, www.rubicon.eu
+//
+// The re-motion Core Framework is free software; you can redistribute it
+// and/or modify it under the terms of the GNU Lesser General Public License
+// as published by the Free Software Foundation; either version 2.1 of the
+// License, or (at your option) any later version.
+//
+// re-motion is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+// GNU Lesser General Public License for more details.
+//
+// You should have received a copy of the GNU Lesser General Public License
+// along with re-motion; if not, see http://www.gnu.org/licenses.
+//
+using Moq;
+using NUnit.Framework;
+using Remotion.Data.DomainObjects.Persistence.Rdbms.Parameters;
+using Remotion.Data.DomainObjects.Persistence.Rdbms.SqlServer.Parameters;
+using Remotion.Data.DomainObjects.Queries;
+using Remotion.Development.UnitTesting;
+
+namespace Remotion.Data.DomainObjects.UnitTests.Persistence.Rdbms.SqlServer.Parameters;
+
+[TestFixture]
+public class SqlDataParameterDefinitionFactoryDecoratorTests
+{
+  [Test]
+  public void DecoratesWith_SqlFulltextDataParameterDefinitionDecorator ()
+  {
+    var queryParameter = new QueryParameter("dummy", "Dummy");
+    var parameterDefinitionStub = new Mock<IDataParameterDefinition>();
+
+    var factoryMock = new Mock<IDataParameterDefinitionFactory>(MockBehavior.Strict);
+    factoryMock.Setup(_ => _.CreateDataParameterDefinition(queryParameter)).Returns(parameterDefinitionStub.Object);
+
+    var factoryDecorator = new SqlDataParameterDefinitionFactoryDecorator(factoryMock.Object);
+
+    var result = factoryDecorator.CreateDataParameterDefinition(queryParameter);
```

Comment:
shouldn't we only add the sqldecorator for type string and char[]?

Reply:
No, because we need the factory-decorator later. However, in the `SqlDataParameterDefinitionFactoryDecorator`, we can only add the Fulltext-decorator for `string or char[]`

Reply:
Oh, yeah, my bad. That's what I meant to write :)